### PR TITLE
Downloading terraform binary version in ocs-ci

### PR DESCRIPTION
**Pull request by vavuthu**
Downloading appropriate terraform binary version in ocs-ci according to ocp release

Signed-off-by: vavuthu <vavuthu@redhat.com>

### CODE REVIEW

File: ocs_ci/deployment/vmware.py [hunk 1]
```diff
@@ -371,6 +371,18 @@ def __init__(self):
...
             self.previous_dir = os.getcwd()
+
+            # Download terraform binary based on ocp version and
+            # update in ENV_DATA
+            # use "0.11.14" for releases below OCP 4.5
+            ocp_version = get_ocp_version()
+            terraform_version = constants.TERRAFORM_OLD_VERSION
+            if Version.coerce(ocp_version) >= Version.coerce('4.5'):
+                terraform_version = config.DEPLOYMENT['terraform_version']
+            terraform_installer = get_terraform(version=terraform_version)
```
Comment (vasukulkarni) [inline ocs_ci/deployment/vmware.py:382]: Can you add a platform check for linux host and skip for others(mac) ? Thanks

Comment (vavuthu) [reply ocs_ci/deployment/vmware.py:382]: For mac, Do you want to skip or download appropriate one ?

Comment (vasukulkarni) [reply ocs_ci/deployment/vmware.py:382]: we should skip here, since its provided by brew.

Comment (vasukulkarni) [reply ocs_ci/deployment/vmware.py:382]: Vijay, thanks for pointing out, but the binary is working fine, we can use the binary for mac too.

Comment (vavuthu) [reply ocs_ci/deployment/vmware.py:382]: downloading the terraform according to platform in https://github.com/red-hat-storage/ocs-ci/pull/2394/files#diff-68bc5ed905b5a4cefbac287573f4f60fR2297

**Comment by vavuthu** [comment]
As expected, terraform took from the bin directory

```
12:20:45 - MainThread - ocs_ci.utility.utils - INFO - Executing command: /home/jenkins/bin/terraform init /home/jenkins/workspace/qe-deploy-ocs-cluster/ocs-ci/external/installer/upi/vsphere/
12:21:17 - MainThread - ocs_ci.utility.utils - INFO - Executing command: /home/jenkins/bin/terraform apply '-var-file=/home/jenkins/current-cluster-dir/openshift-cluster-dir/terraform_data/terraform.tfvars' -auto-approve '/home/jenkins/workspace/qe-deploy-ocs-cluster/ocs-ci/external/installer/upi/vsphere/'
```

File: ocs_ci/deployment/terraform.py [hunk 2]
```diff
@@ -42,9 +55,15 @@ def apply(self, tfvars, bootstrap_complete=False):
...
+            cmd = (
+                f"{self.terraform_installer} apply '-var-file={tfvars}'"
+                f" -auto-approve -var bootstrap_complete=true '{self.path}'"
+            )
         else:
-            cmd = f"terraform apply '-var-file={tfvars}' -auto-approve '{self.path}'"
+            cmd = (
+                f"{self.terraform_installer} apply '-var-file={tfvars}'"
+                f" -auto-approve '{self.path}'"
+            )
```
Comment (petr-balogh) [inline ocs_ci/deployment/terraform.py:66]:
```python
        bootstrap_complete_param ="bootstrap_complete=true " if bootstrap_complete else ""
        cmd = (
            f"{self.terraform_installer} apply '-var-file={tfvars}'"
            f" -auto-approve {bootstrap_complete_param}'{self.path}'"
        )
```

File: ocs_ci/framework/conf/default_config.yaml [hunk 3]
```diff
@@ -70,6 +70,8 @@ DEPLOYMENT:
   preserve_bootstrap_node: False
   # Following options are defaults for the max Noobaa endpoints
   noobaa_endpoints: 2
+  # Terraform version
+  terraform_version: "0.12.26"
```
Comment (petr-balogh) [inline ocs_ci/framework/conf/default_config.yaml:74]: should we put this to OCP 4.5 related config as well?  https://github.com/red-hat-storage/ocs-ci/blob/master/conf/ocp_version/ocp-4.5-config.yaml

Thiking about to have this defined as `default_terraform_version`

File: ocs_ci/deployment/vmware.py [hunk 4]
```diff
@@ -371,6 +371,18 @@ def __init__(self):
...
             )
             self.previous_dir = os.getcwd()
+
+            # Download terraform binary based on ocp version and
+            # update in ENV_DATA
+            # use "0.11.14" for releases below OCP 4.5
+            ocp_version = get_ocp_version()
+            terraform_version = constants.TERRAFORM_OLD_VERSION
+            if Version.coerce(ocp_version) >= Version.coerce('4.5'):
+                terraform_version = config.DEPLOYMENT['terraform_version']
```
Comment (petr-balogh) [inline ocs_ci/deployment/vmware.py:381]: if you will go with default_terraform_version you can do this here
```suggestion
           terraform_version = config.DEPLOYMENT.get('terraform_version', constants.TERRAFORM_OLD_VERSION)
```

File: ocs_ci/deployment/vmware.py [hunk 5]
```diff
@@ -618,6 +630,16 @@ def destroy_cluster(self, log_level="DEBUG"):
         """
         previous_dir = os.getcwd()
 
+        # Download terraform binary based on ocp version and
+        # update in ENV_DATA
+        # use "0.11.14" for releases below OCP 4.5
+        ocp_version = get_ocp_version()
+        terraform_version = constants.TERRAFORM_OLD_VERSION
```
Comment (petr-balogh) [inline ocs_ci/deployment/vmware.py:637]: Write TODO in comment with something like:
```
# TODO: For cluster installed by old version of terraform we need to still run old version for destroy
```
And open issue for this please.

Comment (petr-balogh) [reply ocs_ci/deployment/vmware.py:637]: Ah, I see you've already opened issue here: https://github.com/red-hat-storage/ocs-ci/issues/2399
So you can put it to the comment in TODO here

File: ocs_ci/deployment/terraform.py (hunk 2 above)
Comment (vavuthu) [reply ocs_ci/deployment/terraform.py]: Done

File: ocs_ci/framework/conf/default_config.yaml (hunk 3 above)
Comment (vavuthu) [reply ocs_ci/framework/conf/default_config.yaml:74]: Done

File: ocs_ci/deployment/vmware.py (hunk 4 above)
Comment (vavuthu) [reply ocs_ci/deployment/vmware.py]: As discusses, chnaged on all yaml files

**Pull request merged**